### DISCUSSION TITLE
How to run headless mode while using SB

### DISCUSSION THREAD

**Discussion by randonbr** [seleniumbase/SeleniumBase#3065] [Q&A] [answered]
Hello - I am trying to use headless mode while using SB. Following the guide from:
https://seleniumbase.io/help_docs/uc_mode/#uc-mode

Code:

```
from seleniumbase import SB

with SB(uc=True) as sb:
    url = "https://gitlab.com/users/sign_in"
    sb.uc_open_with_reconnect(url, 4)
```
But when i run this code I get the error:

<img width="1311" alt="Screenshot 2024-08-28 at 12 58 05" src="https://github.com/user-attachments/assets/ea0bc817-f2b7-420a-8d4a-99b5470b79ef">

**Comment by mdmintz** [comment, marked as answer]
Upgrade to the latest version of SeleniumBase. In your screenshot, is says your `yield sb` is on line 885. That's currently on line 995: https://github.com/seleniumbase/SeleniumBase/blob/6c1193e85751236e5d34289901a06a3e37ba84a0/seleniumbase/plugins/sb_manager.py#L995

For UC Mode, make sure you're using Python 3.8 **or newer**. (You were using Python 3.7)

And as mentioned in https://github.com/seleniumbase/SeleniumBase/discussions/2932#discussioncomment-10064354, headless mode is not supported for UC Mode anymore because `PyAutoGUI` requires a headed browser. That's not an issue on Linux if you add `xvfb=True`, which gives you a special virtual display so that you don't need headless mode anymore.
